src/platforms/native/platform.c

```diff
@@ -225,9 +225,11 @@ void platform_init(void)
 	platform_timing_init();
 	cdcacm_init();
 
-	/* On mini hardware, UART and SWD share connector pins.
+	/* On hardware version 1 and 2, UART and SWD share connector pins.
 	 * Don't enable UART if we're being debugged. */
-	if ((platform_hwversion() == 0) || !(SCS_DEMCR & SCS_DEMCR_TRCENA))
+	if ((platform_hwversion() == 0) ||
```

Comment:
the outer pair of parens around each equality suppresses a GCC warning so should be removed. (note: this doesn't apply to the bit mask in the final check which binds looser than || and !)

Reply:
Sounds good. I amended the patch.